truss/templates/shared/logging.py

```diff
@@ -1,63 +1,104 @@
+import json
 import logging
 import sys
+from datetime import datetime
+from enum import Enum
 
+import loguru
 from pythonjsonlogger import jsonlogger
 
 LEVEL: int = logging.INFO
 
-JSON_LOG_HANDLER = logging.StreamHandler(stream=sys.stdout)
-JSON_LOG_HANDLER.set_name("json_logger_handler")
-JSON_LOG_HANDLER.setLevel(LEVEL)
-JSON_LOG_HANDLER.setFormatter(
-    jsonlogger.JsonFormatter("%(asctime)s %(levelname)s %(message)s")
-)
+
+class Lifecycle(Enum):
+    LOAD = "MODEL_LOAD"
+    STARTUP = "MODEL_STARTUP"
+    REQUEST = "REQUEST"
+
+
+def serialize(record):
+    dt = datetime.fromtimestamp(record["time"].timestamp())
+    formatted_time = dt.strftime("%Y-%m-%d %H:%M:%S,%f")[:-3]
+
+    request_id = (
+        str(record["extra"]["request_id"]) if "request_id" in record["extra"] else None
+    )
+    lifecycle = (
+        record["extra"]["lifecycle"].value if "lifecycle" in record["extra"] else None
+    )
+    subset = {
+        "asctime": formatted_time,
+        "message": record["message"],
+        "levelname": record["level"].name,
+        "request_id": request_id,
+        "lifecycle": lifecycle,
+    }
+
+    return json.dumps(subset)
+
+
+def patching(record):
+    # we should be able to label different types of logs here
```

Comment:
what does this mean?

Reply:
got moved into the `serialize` method (where we actually add the labels)